Source/WebKit/Shared/haiku/AuxiliaryProcessMainHaiku.h

```diff
@@ -28,33 +29,114 @@
 #include "AuxiliaryProcess.h"
 #include "WebKit2Initialize.h"
 #include <wtf/RunLoop.h>
+#include <Application.h>
+#include <Looper.h>
+#include <MessageQueue.h>
+#include <String.h>
+#include <map>
+
+using namespace std;
 
 namespace WebKit {
 
-struct AuxiliaryProcessSettings 
-{
+class AuxiliaryProcessMainBase {
+public:
+    virtual bool platformInitialize(char* sig) { return true; }
+    virtual bool parseCommandLine(int argc, char** argv);
+    virtual void platformFinalize() { }
+	
     AuxiliaryProcessInitializationParameters&& takeInitializationParameters() { return WTFMove(m_parameters); }
+
+protected:
     AuxiliaryProcessInitializationParameters m_parameters;
 };
 
+class ProcessApp : public BApplication
+{
+	public:
+	BLooper* messageForward ;
+	status_t result;
+	map<string,BLooper*> proxy;
```

Comment:
what happened to the idea of including the thread_id directly in the messages? it didn't work?

Reply:
I didn't understand how passing pointers will work can you tell me again? In this method I store the looper pointer and uniquely identifiers so it will be helpful in case of multiple web process I guess

Reply:
Instead of AddString("identifier", key) you can do AddPointer("identifier", looper). Then on the way back, you don't need a map.

Reply:
Looper of the sending process?

Reply:
no, the looper the message should be sent to. It needs to be given to the remote application that will send the messages (exactly as it is the case now for the key, right?)